### PR TITLE
Add dependabot for auto updating

### PR DESCRIPTION
**Pull request by tushartathgur**
Adding Dependabot.yml file in order to auto create the PRs for graphing plugins 

Fixes: #189

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: 2
+updates:
+  - package-ecosystem: "npm"
+    directory: "/plugins/grafana-custom-plugins/grafana-chord-plugin"
```
[the remark this answers is not in the document]
Comment (tushartathgur) [reply .github/dependabot.yml:4]: It has to find the plugin.json file for which we have to make it specific to the individual plugin directory

[the remark this answers is not in the document]
Comment (tushartathgur) [reply .github/dependabot.yml:4]: I tried multiple ways for doing so,however, none worked. The package.json file is searched in the directory path and not in the subdirectories which limits us to use the check for the same package in three different directories separately.

**Review by heanlan — COMMENTED** [review]
shall we also add configuration for updating Go dependencies like what Antrea does:
https://github.com/antrea-io/antrea/blob/main/.github/dependabot.yml
https://github.com/antrea-io/antrea/blob/main/.github/workflows/dependabot.yml

Another thing is, I don't see the dependabot.yml in Antrea ever explicitly added `package-ecosystem: "npm"`, but it did check plugins dependencies. Did you have chance to look into why it is the case? I'm fine if we configure it explicitly, as long as it works, just curious if you just happen to know the reason.
https://github.com/antrea-io/antrea/commits/bc3b4452f84f22592afb9b106f9279ddf9712427/plugins/grafana-sankey-plugin/yarn.lock

**Comment by tushartathgur** [comment]
> shall we also add configuration for updating Go dependencies like what Antrea does: https://github.com/antrea-io/antrea/blob/main/.github/dependabot.yml https://github.com/antrea-io/antrea/blob/main/.github/workflows/dependabot.yml
> 
> Another thing is, I don't see the dependabot.yml in Antrea ever explicitly added `package-ecosystem: "npm"`, but it did check plugins dependencies. Did you have chance to look into why it is the case? I'm fine if we configure it explicitly, as long as it works, just curious if you just happen to know the reason. https://github.com/antrea-io/antrea/commits/bc3b4452f84f22592afb9b106f9279ddf9712427/plugins/grafana-sankey-plugin/yarn.lock


Different package-ecosystems will look for vulnerabilities in the specified files, e.g.
- package-ecosystem “go” would look into go.mod and go.sum files.
- package-ecosystem “github-actions” would look into .github/workflows files.
- package-ecosystem “npm” would look into package.json files.

We can add “go” and “GitHub-actions” package-ecosystems, but that will not cover the package.json files in the custom plugins.

File: .github/workflows/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    name: Go tidiness for Dependabot PR
+    # 'dependencies' and 'go' are the default labels used by Dependabot when updating Go dependencies
+    if: ${{ github.actor == 'dependabot[bot]' && contains(github.event.pull_request.labels.*.name, 'dependencies') && contains(github.event.pull_request.labels.*.name, 'go') }}
+    runs-on: [ubuntu-latest]
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        # Check out the pull request HEAD
+        ref: ${{ github.event.pull_request.head.sha }}
+        token: ${{ secrets.ANTREA_BOT_WRITE_PAT }}
```
Comment (heanlan) [inline .github/workflows/dependabot.yml:38]: do you know whether theia has this secret? I have no access to repository secrets so I cannot check. Have you already checked it?

**Comment by heanlan** [comment]
> We can add “go” and “GitHub-actions” package-ecosystems, but that will not cover the package.json files in the custom plugins.

Thanks for adding them. 

The sankey-plugin code was previously in antrea repo, before we moved everything to theia. In antrea repo dependabot.yml, they don't have "npm" package-ecosystems, but the bot also checked plugin dependencies. This link consists of history of the sankey-plugin yarn.lock: https://github.com/antrea-io/antrea/commits/bc3b4452f84f22592afb9b106f9279ddf9712427/plugins/grafana-sankey-plugin/yarn.lock . You'll be able to see multiple PRs created by the bot. It's very strange. I was asking you if you just happen to know the reason. But I think it's fine to have "npm" package-ecosystems in dependabot.yml like what we have in this PR, which makes things clear.

File: .github/workflows/dependabot.yml (hunk 2 above)
Comment (heanlan) [reply .github/workflows/dependabot.yml:38]: I'm a bit confused. Do you mean you created a secret in your forked theia repo? If we would like to merge this PR, shouldn't we create a secret for this theia repo?

Comment (heanlan) [reply .github/workflows/dependabot.yml:38]: I see the documentation: https://docs.github.com/en/code-security/dependabot/working-with-dependabot/configuring-access-to-private-registries-for-dependabot#about-encrypted-secrets-for-dependabot

> When you add a secret at the organization level, you can specify which repositories can access the secret.
Do we know if this secret is set at the organization level and theia repo has access to the secret?

Comment (heanlan) [reply .github/workflows/dependabot.yml:38]: In antrea PRs, we can find the Dependabot workflow, e.g.
https://github.com/antrea-io/antrea/actions/runs/4556122564/jobs/8036076618?pr=4768

but in this PR, we didn't see it showing in the CI checks?

Comment (tushartathgur) [reply .github/workflows/dependabot.yml:38]: Sorry for the confusion, what I meant is for theia we right now do not have a PAT, we can use this for right now as a placeholder, but it won’t do anything as it is only present for Antrea, once a new PAT becomes available for Theia, we will add that in ANTREA_BOT_WRITE_PAT’s place.

**Comment by tushartathgur** [comment]
> > We can add “go” and “GitHub-actions” package-ecosystems, but that will not cover the package.json files in the custom plugins.
> 
> Thanks for adding them.
> 
> The sankey-plugin code was previously in antrea repo, before we moved everything to theia. In antrea repo dependabot.yml, they don't have "npm" package-ecosystems, but the bot also checked plugin dependencies. This link consists of history of the sankey-plugin yarn.lock: https://github.com/antrea-io/antrea/commits/bc3b4452f84f22592afb9b106f9279ddf9712427/plugins/grafana-sankey-plugin/yarn.lock . You'll be able to see multiple PRs created by the bot. It's very strange. I was asking you if you just happen to know the reason. But I think it's fine to have "npm" package-ecosystems in dependabot.yml like what we have in this PR, which makes things clear.

Dependabot can create PRs in two ways, one is for security vulnerabilities and other is version vulnerabilities. For version vulnerabilities, we would require a dependabot.yml file however for security vulnerabilities, PRs can be generated automatically without the need of dependabot.yml, like the ones we can see right now in Theia PRs.

**Comment by elton-furtado** [comment]
> > > We can add “go” and “GitHub-actions” package-ecosystems, but that will not cover the package.json files in the custom plugins.
> > 
> > 
> > Thanks for adding them.
> > The sankey-plugin code was previously in antrea repo, before we moved everything to theia. In antrea repo dependabot.yml, they don't have "npm" package-ecosystems, but the bot also checked plugin dependencies. This link consists of history of the sankey-plugin yarn.lock: https://github.com/antrea-io/antrea/commits/bc3b4452f84f22592afb9b106f9279ddf9712427/plugins/grafana-sankey-plugin/yarn.lock . You'll be able to see multiple PRs created by the bot. It's very strange. I was asking you if you just happen to know the reason. But I think it's fine to have "npm" package-ecosystems in dependabot.yml like what we have in this PR, which makes things clear.
> 
> Dependabot can create PRs in two ways, one is for security vulnerabilities and other is version vulnerabilities. For version vulnerabilities, we would require a dependabot.yml file however for security vulnerabilities, PRs can be generated automatically without the need of dependabot.yml, like the ones we can see right now in Theia PRs.

Is there any documentation that shows the difference the behavior for these two types of vulnerabilities?

**Comment by tushartathgur** [comment]
> > > > We can add “go” and “GitHub-actions” package-ecosystems, but that will not cover the package.json files in the custom plugins.
> > > 
> > > 
> > > Thanks for adding them.
> > > The sankey-plugin code was previously in antrea repo, before we moved everything to theia. In antrea repo dependabot.yml, they don't have "npm" package-ecosystems, but the bot also checked plugin dependencies. This link consists of history of the sankey-plugin yarn.lock: https://github.com/antrea-io/antrea/commits/bc3b4452f84f22592afb9b106f9279ddf9712427/plugins/grafana-sankey-plugin/yarn.lock . You'll be able to see multiple PRs created by the bot. It's very strange. I was asking you if you just happen to know the reason. But I think it's fine to have "npm" package-ecosystems in dependabot.yml like what we have in this PR, which makes things clear.
> > 
> > 
> > Dependabot can create PRs in two ways, one is for security vulnerabilities and other is version vulnerabilities. For version vulnerabilities, we would require a dependabot.yml file however for security vulnerabilities, PRs can be generated automatically without the need of dependabot.yml, like the ones we can see right now in Theia PRs.
> 
> Is there any documentation that shows the difference the behavior for these two types of vulnerabilities?

Yes, please find the documentation below 
https://docs.github.com/en/code-security/dependabot/dependabot-security-updates/about-dependabot-security-updates

**Comment by heanlan** [comment]
Are we ready to merge this PR?

**Comment by tushartathgur** [comment]
> Are we ready to merge this PR?
> 
> Please add `Fixes: #189` in the PR description.

Yes, This PR is ready to merge.

**Review by heanlan — APPROVED** [review]

**Pull request merged**